### DISCUSSION TITLE
duckdb do not support Oracle database

### DISCUSSION THREAD

**Discussion by moukexian** [duckdb/duckdb#10928] [Ideas]
### What happens?

可惜不能支持 oracle
can not operate like "attach oracle..."

### To Reproduce

目前有大量的行业用户使用 oracle 数据库。 可惜 duckdb 不能支持 oracle.
can not operate like "attach oracle..."

### OS:

windows

### DuckDB Version:

10

### DuckDB Client:

python

### Full Name:

moukexian

### Affiliation:

oracle

### Have you tried this on the latest [nightly build](https://duckdb.org/docs/installation/?version=main)?

I have tested with a nightly build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by szarnyasg** [comment]
Hi, this is expected: DuckDB currently has extensions to support MySQL, PostgreSQL, and SQLite.

**Reply by moukexian** [reply]
Does duckdb have a plan to support Oracle?
I really hope that Duckdb has the same support for Oracle as it does for MySQL databases. Can use "attach oracle..."

**Comment by moukexian** [comment]
Does duckdb have a plan to support Oracle?

**Comment by moukexian** [comment]
I really hope that Duckdb has the same support for Oracle as it does for MySQL databases. Can use "attach oracle..."

**Comment by szawadski** [comment]
@szarnyasg : Is there any plan to have an extension that would allow to connect to other DB types than the currently supported ones with for instance a standard interface like JDBC?

**Comment by ptrxyz** [comment]
I also would like to see this! However I understand that oracle db connector is quite a beast... :+1:

**Comment by chris-dailey** [comment]
I too would love to see this feature added.

**Comment by visuman** [comment]
Any plan on implementing oracle support?

**Comment by szawadski** [comment]
Hello,

I think that considering the ongoing development of duckdb airport
extension, it should be possible to connect duckdb to oracle soon without
the need to develop a specific interface.
[image: image.png]
Have a look at the video.
https://youtu.be/-AfgEiE2kaI?si=Eqf1rNiHKrtkUCYW

Regards,

Sébastien

Le mer. 19 mars 2025 à 13:57, Vishnu Suman ***@***.***> a
écrit :

**Reply by pedroarthuralvesdeoliveira** [reply]
Any updates on this?

**Reply by chris-dailey** [reply]
The airport extension can be installed / loaded from the community repository as of DuckDB version 1.3.2. However, an Apache Arrow Flight server coded to implement airport is still needed for connectivity to Oracle (or any other databases without core extensions).

**Comment by rustyconover** [comment]
If you’re interested in an Oracle connector reach out to me at
***@***.***

**Reply by chris-dailey** [reply]
@rustyconover, I'm highly interested in an Oracle connector for DuckDB. However, the e-mail address in your message shows as asterisks. I'm not sure if you shared it briefly, but I only had e-mail notifications enabled at the time and the message showed asterisks there too. I've since enabled GitHub notifications and made my e-mail address public if you would prefer to reach out to me instead.

**Reply by pedroarthuralvesdeoliveira** [reply]
> ####

How are you dealing with data from Oracle within Duckdb? Are you parsing to csv and then querying it from duckdb or something else?

**Comment by staticlibs** [comment]
It is now possible to access Oracle using [odbc_scanner extension](https://github.com/duckdb/odbc-scanner).

**Reply by chris-dailey** [reply]
@staticlibs, Thank you for the update. Would it be possible to create something that works similarly to the core extensions `MySQL` / `PostgreSQL` / `SQLite` for Oracle?

**Reply by staticlibs** [reply]
@chris-dailey 

In addition to `mysql_query`/`postgres_query`/`sqlite_query` functions (that are covered by `odbc_query` for Oracle) the scanners also support attaching the remote DB to the DuckDB catalog. This is currently not supported in `odbc_scanner`, but is planned for future when catalog attach API is added to stable DuckDB C API.

**Reply by rustyconover** [reply]
Is ADBC planned as well?

**Reply by staticlibs** [reply]
@rustyconover 

AFAIK no, there are thoughts about JDBC and ADBC scanners, but none of them are planned now.

**Reply by rustyconover** [reply]
Ok, please let me know if that changes.

**Reply by rustyconover** [reply]
> @rustyconover
> 
> AFAIK no, there are thoughts about JDBC and ADBC scanners, but none of them are planned now.

I cranked out a pretty slim ADBC extension tonight modeled after `odbc_scanner`

https://github.com/Query-farm/adbc_scanner

I'll polish and ship in the next few days.
